src/Jackett.Common/Services/UpdateService.cs

```diff
@@ -91,59 +96,85 @@ private async Task CheckForUpdates()
                 logger.Info("Skipping checking for new releases as the debugger is attached.");
                 return;
             }
-            var currentVersion = EnvironmentUtil.JackettVersion();
-            if (currentVersion == "v0.0.0")
+
+            var currentVersion = ParseVersion(EnvironmentUtil.JackettVersion());
+
+            if (currentVersion == new Version(0, 0, 0))
             {
                 logger.Info("Skipping checking for new releases because Jackett is runing in IDE.");
                 return;
             }
 
-            var isWindows = Environment.OSVersion.Platform != PlatformID.Unix;
+            var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
 
             var trayIsRunning = false;
             if (isWindows)
+            {
                 trayIsRunning = Process.GetProcessesByName("JackettTray").Length > 0;
+            }
 
             try
             {
-                var response = await client.GetResultAsync(new WebRequest()
+                var response = await client.GetResultAsync(new WebRequest
                 {
                     Url = "https://api.github.com/repos/Jackett/Jackett/releases",
                     Encoding = Encoding.UTF8,
                     EmulateBrowser = false
                 });
 
                 if (response.Status != System.Net.HttpStatusCode.OK)
-                    logger.Error($"Failed to get the release list: {response.Status}");
+                {
+                    logger.Error("Failed to get the release list: {0}", response.Status);
+                }
 
                 var releases = JsonConvert.DeserializeObject<List<Release>>(response.ContentString);
 
                 if (!serverConfig.UpdatePrerelease)
+                {
                     releases = releases.Where(r => !r.Prerelease).ToList();
+                }
 
-                if (releases.Count > 0)
+                if (releases.Any())
                 {
                     var latestRelease = releases.OrderByDescending(o => o.Created_at).First();
-                    if (latestRelease.Name != currentVersion)
```

Comment:
wouldn't it be easier to fetch the latest version number from `https://api.github.com/repos/Jackett/Jackett/releases/latest` instead of sorting the releases page?